File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
import sys
sys.setrecursionlimit(10 ** 9)

def recursive(s):
    if not s:
        return True
    
    stk = []
    isNothing = True
    for i, c in enumerate(s):
        if not stk:
            stk.append(c)
            continue
        
        if stk[-1] == c:
            isNothing = False
            stk.pop()
            continue
            
        stk.append(c)
    
    if isNothing:
        return False
    
    return recursive(stk)

def solution(s):
    return int(recursive(list(s)))

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s){
    let stk = [];
    for(let v of s){
        if (stk.at(-1) === v){
            stk.pop();
            continue;
        } 
        
        stk.push(v)
    }
    
    return stk.length === 0 ? 1 : 0;
}